src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -104,6 +104,8 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
             marshaller.marshal(toSave, fileWriter);
 
+        } catch (FileNotFoundException fnfe) {
```

Comment:
The exercise require you to explicitly check if a file exisits or not and then throw the exception in the save() method. 

The best place to catch and terminate is Main::executeCommand() (this method already has a catch block to terminate upon exception.